src/activity_item_handling.cpp

```diff
@@ -1187,77 +1229,119 @@ static activity_reason_info can_do_activity_there( const activity_id &act, playe
     // see activity_handlers.h cant_do_activity_reason enums
     zone_manager &mgr = zone_manager::get_manager();
     std::vector<zone_data> zones;
-    if( act == activity_id( "ACT_VEHICLE_DECONSTRUCTION" ) ) {
-        if( g->m.getlocal( g->u.activity.placement ) == src_loc ) {
-            return activity_reason_info::fail( ALREADY_WORKING );
-        }
+    if( act == activity_id( "ACT_VEHICLE_DECONSTRUCTION" ) ||
+        act == activity_id( "ACT_VEHICLE_REPAIR" ) ) {
+        std::vector<int> already_working_indexes;
         for( const npc &guy : g->all_npcs() ) {
-            if( g->m.getlocal( guy.activity.placement ) == src_loc || guy.pos() == src_loc ) {
+            if( guy.disp_name() == p.disp_name() ) {
```

Comment:
What actually happens when the player has two followers with the same name? Should compare the pointers directly. (Comparing pointer to npc and pointer to player will work fine.)

Reply:
Ah hah, didnt know if comparing npc to player would work.  That makes more sense. thanks.